core/dbt/graph/selector.py

```diff
@@ -87,12 +87,15 @@ def get_nodes_from_criteria(
             )
             return set(), set()
 
+        neighbors = self.collect_specified_neighbors(spec, collected)
+        selected = collected | neighbors
+
+        # if --indirect-selection EMPTY, do not expand to adjacent tests
         if spec.indirect_selection == IndirectSelection.Empty:
-            return collected, set()
+            return selected, set()
```

Comment:
This looks like a changed behavior. But after some digging in I think `IndirectSelection.Empty` should actually only affect expand_selection not collect neighbors?

I looked at the [PR](https://github.com/dbt-labs/dbt-core/pull/7008/files#diff-e7dc35d6a37ff0275f71675e8a0d2364827bdd88df814aed971c7b2946d0b253) introduced this logic and found the tests introduced there doesn't always have empty neighbors. @aranke why were we only returning `collected` previously?

Reply:
@ChenyuLInx You're right - I believe this change fixes a defect in the current code. I can open a separate issue/PR if that feels cleaner.

`collect_specified_neighbors` is our method for adding children/parents (detected via the `+` operator), and right now `--indirect-selection empty` completely ignores them!

I've updated the unit test so that it fails without this change.